sleap/nn/tracking.py

```diff
@@ -443,7 +600,11 @@ def _init_matching_queue(self):
         return deque(maxlen=self.track_window)
 
     def reset_candidates(self):
-        self.track_matching_queue = deque(maxlen=self.track_window)
+        if self.max_tracking and self.track_matching_queue_dict:
+            for track, candidates in self.track_matching_queue_dict.items():
+                candidates = deque(maxlen=self.track_window)
+        else:
+            self.track_matching_queue = deque(maxlen=self.track_window)
```

Comment:
Set the item in `track_matching_queue` itself, this might not actually update the `track_matching_queue`
```python
In [1]: d = {1: 1}

In [2]: for k, v in d.items():
   ...:     v = 3
   ...:

In [3]: d
Out[3]: {1: 1}
```